source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -732,6 +732,54 @@ def _tag_by_mode(responses):
             _tag_journey_by_mode(j)
 
 
+def tag_robust_journeys(responses):
+    for r in responses:
+        for j in r.journeys:
+            if is_robust_journey(j):
+                j.tags.append("robust")
+
+
+robust_physical_modes = [
+    "physical_mode:RapidTransit",
+    "physical_mode:Metro",
+    "physical_mode:Train",
+    "physical_mode:RailShuttle",
+    "physical_mode:LocalTrain",
+    "physical_mode:LongDistanceTrain",
+]
+robust_fallback_modes = [response_pb2.Bike, response_pb2.Walking]
+# returns true if :
+#  - a journey has at least one public transport section
+#  - all public transport sections use robust physical modes
+#  - start and end fallbacks use robust fallback mode
+def is_robust_journey(journey):
+    found_a_robust_mode = False
+    found_a_non_robust_fallback = False
+    for section in journey.sections:
+
+        # check if this section uses a non robust fallback mode
+        if section.HasField("street_network"):
+            street_network = section.street_network
+            if street_network.HasField("mode"):
+                mode = street_network.mode
+                if mode not in robust_fallback_modes:
+                    found_a_non_robust_fallback = True
+                    break
```

Comment:
How about doing `return False` from here and not use the `found_a_non_robust_fallback` at all?

Reply:
because then a direct path would be "robust" and we want a journey to be "robust" only if it contains a public transport section